code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -528,26 +537,12 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		else if(rule.type == "Midround")
 			midround_rules = remove_from_list(midround_rules, rule.type)
 
-	if (rule.execute())
-		log_game("DYNAMIC: Injected a [rule.ruletype == "latejoin" ? "latejoin" : "midround"] ruleset [rule.name].")
-		spend_threat(rule.cost)
-		threat_log += "[worldtime2text()]: [rule.ruletype] [rule.name] spent [rule.cost]"
-		if(rule.flags & HIGHLANDER_RULESET)
-			highlander_executed = TRUE
-		else if(rule.flags & ONLY_RULESET)
-			only_ruleset_executed = TRUE
-		if(rule.ruletype == "Latejoin")
-			var/mob/M = pick(rule.candidates)
-			message_admins("[key_name(M)] joined the station, and was selected by the [rule.name] ruleset.")
-			log_game("DYNAMIC: [key_name(M)] joined the station, and was selected by the [rule.name] ruleset.")
-		executed_rules += rule
-		rule.candidates.Cut()
-		if (rule.persistent)
-			current_rules += rule
+	if(rule.delay > 0)
```

Comment:
I think you could also just remove this check and make everything just use timers even with zero delay.

Reply:
Its true that we could, but there wouldn't be a point as far as I can see. First of all it would make us lose the return values of anything we timer. So to have any chance of reverting failed executes, we'd then have to process and check them until started properly. Maybe some day, but its not for this PR.

Reply:
Whatever, I went with this.